File: app/src/main/java/pl/szczodrzynski/edziennik/datamodels/MessageRecipientFull.java
package pl.szczodrzynski.edziennik.datamodels;

public class MessageRecipientFull extends MessageRecipient {
    public String fullName = null;

    public MessageRecipientFull(int profileId, long id, long replyId, long readDate, long messageId) {
        super(profileId, id, replyId, readDate, messageId);
    }

    public MessageRecipientFull(int profileId, long id, long messageId) {
        super(profileId, id, messageId);
    }

    public MessageRecipientFull() {
        super();
    }
}

File: app/src/main/java/pl/szczodrzynski/edziennik/data/db/modules/messages/MessageRecipientFull.kt
package pl.szczodrzynski.edziennik.data.db.modules.messages

import androidx.room.Ignore
import pl.szczodrzynski.edziennik.fixName

class MessageRecipientFull : MessageRecipient {
    var fullName: String? = ""
        get() {
            return field?.fixName() ?: ""
        }

    @Ignore
    constructor(profileId: Int, id: Long, replyId: Long, readDate: Long, messageId: Long) : super(profileId, id, replyId, readDate, messageId) {}
    @Ignore
    constructor(profileId: Int, id: Long, messageId: Long) : super(profileId, id, messageId) {}
    constructor() : super() {}
}